parser/charset/charset.go

```diff
@@ -91,8 +91,8 @@ func GetSupportedCharsets() []*Charset {
 	}
 
 	// sort charset by name.
-	slices.SortFunc(charsets, func(i, j *Charset) bool {
```

Comment:
Now tidb use go1.21. we can use std package ```slices``` to replace the ```exp/slices```.

Reply:
Should we also require `go1.21` for `github.com/pingcap/tidb/parser`?
It can be used outside `tidb` and is currently has `go 1.18` in [go.mod](https://github.com/pingcap/tidb/blob/master/parser/go.mod)

Reply:
considering parser may be publicly used by other applications, I prefer we don't upgrade golang versions too aggressively. my opinion is parser module should use the second latest major version of golang. @hawkingrei